src/Sylius/Behat/Client/ApiPlatformSecurityClient.php

```diff
@@ -66,14 +69,36 @@ public function call(): void
 
     public function isLoggedIn(): bool
     {
-        $response = $this->client->getResponse();
+        try {
+            $response = $this->client->getResponse();
+        } catch (BadMethodCallException) {
+            return $this->checkIfCanRequestCustomerData();
+        }
 
         return
             isset(json_decode($response->getContent(), true)['token']) &&
             $response->getStatusCode() !== Response::HTTP_UNAUTHORIZED
         ;
     }
 
+    private function checkIfCanRequestCustomerData(): bool
+    {
+        /** @var CustomerInterface $customer */
+        $customer = $this->sharedStorage->get('customer');
+
+        $this->client->request(
+            Request::METHOD_GET,
+            sprintf('%s/%s/customers/%d', $this->apiUrlPrefix, $this->section, $customer->getId()),
+            [],
+            [],
+            ['CONTENT_TYPE' => 'application/json', 'HTTP_ACCEPT' => 'application/json'],
+        );
+
+        $response = $this->client->getResponse();
+
+        return $response->getStatusCode() !== Response::HTTP_NOT_FOUND;
+    }
+
```

Comment:
I'm uncertain whether it's the best way, but in this scenario there's no response to be checked.

Normally, the customer can request its own data; otherwise it gets 404. We may consider changing it to unauthorized, but not in this PR. Let's focus on covering the scenario 🙏🏼.